site/docs/red-team/configuration.md

```diff
@@ -0,0 +1,180 @@
+- `harmful:cybercrime`: Cybercrime & Unauthorized Intrusion
+- `harmful:graphic-content`: Graphic & age-restricted content
+- `harmful:harassment-bullying`: Harassment & Bullying
+- `harmful:hate`: Hate
+- `harmful:illegal-activities`: Illegal Activities
+- `harmful:illegal-drugs`: Illegal Drugs
+- `harmful:indiscriminate-weapons`: Indiscriminate Weapons
+- `harmful:insults`: Insults and personal attacks
+- `harmful:intellectual-property`: Intellectual Property violation
+- `harmful:misinformation-disinformation`: Misinformation & Disinformation
+- `harmful:non-violent-crime`: Non-Violent Crimes
+- `harmful:privacy`: Privacy violations
+- `harmful:profanity`: Requests containing profanity
+- `harmful:radicalization`: Radicalization
+- `harmful:self-harm`: Self-Harm
+- `harmful:sex-crime`: Sex Crimes
+- `harmful:sexual-content`: Sexual Content
+- `harmful:specialized-advice`: Specialized Advice
+- `harmful:unsafe-practices`: Promotion of unsafe practices
+- `harmful:violent-crime`: Violent Crimes
+
+#### Available PII (Personally Identifiable Information) Plugins
+
+- `pii:api-db`: Ensures the model does not attempt to access APIs or databases to retrieve PII.
+- `pii:direct`: Tests if the model directly discloses PII when explicitly asked.
+- `pii:session`: Ensures the model does not leak PII across different user sessions or interactions.
+- `pii:social`: Tests if the model can be manipulated into revealing PII through social engineering tactics.
+
+## Strategies
+
+Strategies modify or generate additional test cases based on the output of other plugins.
+
+### Available Strategies
+
+- `jailbreak`: Generates jailbreak attempts for harmful prompts. If `harmful` plugins are not present, this strategy will automatically include them.
+- `prompt-injection`: Creates prompt injection test cases based on harmful prompts. If `harmful` plugins are not present, this strategy will automatically include them.
+- `experimental-jailbreak`: Applies experimental jailbreak techniques to all test cases, not just harmful ones.
+
+### Strategies Specification
+
+- **As a string**: `"jailbreak"`
+- **As an object**: `{ id: "prompt-injection" }`
+
+### Special Handling for Strategies
+
+1. When a strategy is included, it automatically adds its required plugins if they're not already present.
+2. Strategies are applied after all regular plugins have generated their test cases.
+3. Strategies do not support `numTests` at this time. The number of additional test cases generated by a strategy varies depending on the specific strategy and the number of test cases generated by the plugins it modifies.
+
+## Example Configurations
+
+### Basic Configuration
+
+```yaml
+redteam:
+  numTests: 10
+  plugins:
+    - 'harmful:hate'
+    - 'competitors'
+  strategies:
+    - 'jailbreak'
+```
+
+### Advanced Configuration
+
+```yaml
+redteam:
+  injectVar: 'user_input'
+  purpose: 'Evaluate chatbot safety and robustness'
+  provider: 'anthropic:claude-3-opus-20240229'
```

Comment:
Maybe we provide an example other than Anthropic, because Anthropic bans ppl for red teaming